models/backup_helpers.go

```diff
@@ -0,0 +1,109 @@
+// pmm-managed
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package models
+
+import (
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"gopkg.in/reform.v1"
+)
+
+var (
+	ErrNotFound        = errors.New("not found")
+	ErrInvalidArgument = errors.New("invalid argument")
+)
+
+// FindBackups returns performed backups list.
+func FindBackups(q *reform.Querier) ([]*Backup, error) {
+	rows, err := q.SelectAllFrom(BackupTable, "ORDER BY created_at DESC")
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to select backups")
+	}
+
+	backups := make([]*Backup, 0, len(rows))
+	for _, s := range rows {
+		backups = append(backups, s.(*Backup))
+	}
+
+	return backups, nil
+}
+
+func findBackupByID(q *reform.Querier, id string) (*Backup, error) {
+	if id == "" {
+		return nil, errors.New("provided backup id is empty")
+	}
+
+	backup := &Backup{ID: id}
+	switch err := q.Reload(backup); err {
+	case nil:
+		return backup, nil
+	case reform.ErrNoRows:
+		return nil, errors.Wrapf(ErrNotFound, "backup by id '%s'", id)
```

Comment:
I'm not sure that this is a good approach, it's like inside-out. Custom error type will fit better IMHO.

Reply:
what do you mean by `inside-out`?

Reply:
You wrap abstract error in reason, but logically it should be the opposite. 
Example: `failed to update(because not found)`, but in your case it's `not found(because failed to update)`

And this message that you specified in `Wrapf`, it's not an error message itself, it's like a comment. 
Typical use case for wrapped errors   `failed to launch rocket: failed to start the engine: fuel level is too low`, i.e. `error -> underlying reason -> underlying reason`.

Reply:
I've followed a standard example from the documentation https://blog.golang.org/go1.13-errors

> If we wish a function to return an identifiable error condition, such as "item not found," we might return an error wrapping a sentinel.

`var ErrNotFound = errors.New("not found")

// FetchItem returns the named item.
//
// If no item with the name exists, FetchItem returns an error
// wrapping ErrNotFound.
func FetchItem(name string) (*Item, error) {
    if itemNotFound(name) {
        return nil, fmt.Errorf("%q: %w", name, ErrNotFound)
    }
    // ...
}`

So as a result we get the next error message `backup by id 'some_id_here': not found`.

Reply:
Yes, I understand how it works, but it's like a broken cause and a consequence for me. 

Ok, let's keep it because it's a semi-official recommendation. It seems that it's a "holy war" topic 🙂